Tests/Shared.Specs/ExceptionAssertionSpecs.cs

```diff
@@ -895,6 +894,62 @@ public void When_no_exception_should_be_thrown_and_none_was_it_should_not_throw(
             //-----------------------------------------------------------------------------------------------------------
             foo.Invoking(f => f.Do()).Should().NotThrow();
         }
+
+        private static void ThrowUntil(Stopwatch watch, int throwTime)
+        {
+                if(watch.ElapsedMilliseconds <= throwTime)
+                    throw new ArgumentException("An exception was forced");
+        }
+
+        [Fact]
+        public void When_no_exception_should_be_thrown_after_wait_time_but_it_was_it_should_throw()
+        {
+            //-----------------------------------------------------------------------------------------------------------
+            // Arrange
+            //-----------------------------------------------------------------------------------------------------------
+            var foo = A.Fake<IFoo>();
+            var watch = Stopwatch.StartNew();
+            var waitTime = 100;
+            var pollInterval = 10;
+
+            void ThrowLongerThanWaitTime() => ThrowUntil(watch, waitTime + waitTime / 2);
+
+            A.CallTo(() => foo.Do()).Invokes(ThrowLongerThanWaitTime);
+            //-----------------------------------------------------------------------------------------------------------
+            // Act
+            //-----------------------------------------------------------------------------------------------------------
+            Action action = () => foo.Invoking(f => f.Do())
+                                     .Should().NotThrowAfter(waitTime, pollInterval, "we passed valid arguments");
+
+            //-----------------------------------------------------------------------------------------------------------
+            // Assert
+            //-----------------------------------------------------------------------------------------------------------
+            action
+                .Should().Throw<XunitException>()
+                         .Where(e => e.Message.Contains(
```

Comment:
🔧 Use `WithMessage` with wildcards as the test should not depend on how FakeItEasy name their exceptions.
E.g. 
```csharp
action.Should().Throw<XunitException>()
    .WithMessage("*we passed valid arguments*");
```